python-sdk/src/astro/databases/snowflake.py

```diff
@@ -393,12 +392,12 @@ def create_stage(
         stage.set_url_from_file(file)
 
         fileformat = ASTRO_SDK_TO_SNOWFLAKE_FILE_FORMAT_MAP[file.type.name]
-        copy_options = [COPY_OPTIONS[file.type.name]]
+        copy_options = [COPY_OPTIONS.get(file.type.name)] if COPY_OPTIONS.get(file.type.name) else []
```

Comment:
Won't we face the same issue with `fileformat`?

Reply:
Shouldn't we error out, if the file type is not supported?

Reply:
@feluelle I don't think this line's matter is of the filetype being supported, but we're defining default values for COPY OPTIONS for some of the filetypes.

@sunank200 Are we making it clear in the documentation that we are setting these values as defaults for these specific filetypes? It may be worth at least having a link to the code if we don't want to duplicate the information.

Do you remember why we added these defaults? I believe it was due to real datasets, but it may be worth adding some comments as well.

Reply:
@tatiana one option is to completely remove all the `copy_options` by default and the user can pass in values as required depending on the type of file as they do in the `COPY INTO` command. WDYT?. Currently, users cannot override `MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE` for `NDJSON` and `PARQUET` files as its set as default values.

Reply:
@sunank200 I think it is a good solution - not to have defaults and allow users to specify them. It feels wrong to decide on the user's behalf without allowing them to override.

There was a reason this was done in the past - probably due to real datasets. What happens to our tests if we remove these defaults? What happens to the benchmark script if we move them?

Reply:
I have removed all the default `copy_options`.

Reply:
All test work without default `copy_options`.